src/token/api/src/canister/is20_transactions.rs

```diff
@@ -15,9 +15,11 @@ use super::TokenCanisterAPI;
 /// transaction will fail with `TxError::AmountTooSmall` error.
 pub fn transfer_include_fee(
     canister: &impl TokenCanisterAPI,
-    caller: CheckedPrincipal<WithRecipient>,
+    from: TokenHolder,
+    to: TokenReceiver,
```

Comment:
Here and in other methods we replaced the `CheckedPrincipal` with unchecked `AccountIdentifier`. Is there a reason to remove the static check that the `CheckedPrincipal` provides?

Reply:
The`CheckPrincipal` is being used at the trait api, Instead of passing the caller args, I use the field of the `caller` to parse the `Principal` and the `SubAccount` into `AccountIdentifier`.

Reply:
The whole idea of the `CheckedPrincipal` is to be used where the business logic is. Otherwise there's no point in it. It provides some compile-time guarantees for the business logic that your code is lacking.

For example, I can call your version of the  `transfer_include_fee` as
```
transfer_include_fee(canister, to, from, amount)
```
which will break the business logic, but the compiler will still be happy. And this kind of error is quite difficult to find.

Reply:
Yes, that makes sense.